code/go/0chain.net/smartcontract/minersc/fees.go

```diff
@@ -19,50 +20,37 @@ import (
 )
 
 func (msc *MinerSmartContract) activatePending(mn *MinerNode) {
-	for id, pool := range mn.Pending {
-		pool.Status = ACTIVE
-		mn.Active[id] = pool
-		mn.TotalStaked += int64(pool.Balance)
-		delete(mn.Pending, id)
+	for _, pool := range mn.Pools {
+		if pool.Status == stakepool.Pending {
+			pool.Status = stakepool.Active
+			mn.NumPending--
```

Comment:
I don't see code that increase the `NumPending`, I guess it needs to be updated when the pool is added, not sure about other places. Anyway, please make sure the `NumPending` is updated correctly when the `pool.Status` is changed. So as the `NumActive`. 

Write unit tests to make sure that NumPending and NumActive are updated appropriately if possible.